bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoGatewayHandler.java

```diff
@@ -0,0 +1,86 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.miio.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.miio.internal.basic.BasicChannelTypeProvider;
+import org.openhab.binding.miio.internal.basic.MiIoDatabaseWatchService;
+import org.openhab.binding.miio.internal.cloud.CloudConnector;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.BridgeBuilder;
+import org.openhab.core.thing.type.ChannelTypeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MiIoGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+@NonNullByDefault
+public class MiIoGatewayHandler extends MiIoBasicHandler implements BridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MiIoGatewayHandler.class);
+
+    public MiIoGatewayHandler(Bridge thing, MiIoDatabaseWatchService miIoDatabaseWatchService,
+            CloudConnector cloudConnector, ChannelTypeRegistry channelTypeRegistry,
+            BasicChannelTypeProvider basicChannelTypeProvider, TranslationProvider i18nProvider,
+            LocaleProvider localeProvider) {
+        super(thing, miIoDatabaseWatchService, cloudConnector, channelTypeRegistry, basicChannelTypeProvider,
+                i18nProvider, localeProvider);
+    }
+
+    @Override
+    public Bridge getThing() {
+        return (Bridge) super.getThing();
+    }
+
+    /**
+     * Creates a bridge builder, which allows to modify the bridge. The method
+     * {@link BaseThingHandler#updateThing(Thing)} must be called to persist the changes.
+     *
+     * @return {@link BridgeBuilder} which builds an exact copy of the bridge
+     */
+    @Override
+    protected BridgeBuilder editThing() {
+        return BridgeBuilder.create(thing.getThingTypeUID(), thing.getUID()).withBridge(thing.getBridgeUID())
+                .withChannels(thing.getChannels()).withConfiguration(thing.getConfiguration())
+                .withLabel(thing.getLabel()).withLocation(thing.getLocation()).withProperties(thing.getProperties());
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        logger.info("Child registered with gateway: {}  {} -> {} {}", childThing.getUID(), childThing.getLabel(),
+                getThing().getUID(), getThing().getLabel());
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug. See https://next.openhab.org/docs/developer/guidelines.html#f-logging

Same for below.